.github/workflows/api-spec-lint.yaml

```diff
@@ -0,0 +1,36 @@
+name: "Confirm generated file changes are checked in"
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - devnet
+      - testnet
+      - auto
+      - canary
+
+env:
+  CARGO_INCREMENTAL: "0"
+  CARGO_TERM_COLOR: always
+
+# Cancel redundant builds
+concurrency:
+  # Cancel redundant builds on PRs (only on PR, not on branches)
+  group: ${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.ref) || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  # Confirm that any changes to the API specs have been checked in.
+  api-spec-check:
```

Comment:
honestly, it'd just add this job to the existing `lint-test.yaml` workflow.

Reply:
I believe Wolfgang is adding another similar check, perhaps we could have a "ensure-changes-checked-in.yaml" kind of thing?